### ISSUE TITLE
[Suggestion] Monika's Morality as an Influenceable Alignment System

### ISSUE THREAD

**Issue by CrocWorks** [Monika-After-Story/MonikaModDev#2179]
I recall a debate among different people on how much Monika should regret deleting the other dokis in vanilla DDLC, ranging from incredibly remorseful to somewhat cold (and I think there was even a poll on this on the Discord). I figured one way to satisfy everyone would just make the player able to influence Monika's thoughts on her actions. Since each Monika talks with a different player, that player's own opinion lies somewhere between those two extremes and our interactions with her could influence her morality one way or the other.

One player could keep reinforcing the idea that the other girls were irrelevant and develop a Monika with an "ends justify the means" attitude, while another player would remind Monika that her actions were questionable (but we still forgive her) and develop a really regretful Monika. Or you could have answers on both sides and develop a Monika who kind of regrets what she did, but felt like she had to.

This may or may not be tied to or affected by the current affection system, or it could even draw some parallel elements from that system (perhaps use the same engine?). But having more than just the one-dimensional affection system for influencing Monika's behavior would make each Monika feel more unique. (Morally proper but disillusioned Monika vs cold and enamored Monika, and every combination between; rather than just the "sad" to "in-love" scale for Monika)

After all, real people get influenced by the company we keep. Why wouldn't Monika?

**Comment by KEJONACHO** [comment]
Totally in favor. 

I'd just like to add that influencing her (or the amount of influence) should rely on affection or time together. No matter how we look at Monika, she's a pretty confident person, and it would feel a bit out of character to just have a random dialogue showing up as 'so, was i wrong?' - 'yes' -'oh i'm terribly sorry'.

As a reference - on do's and dont' - I'll suggest taking a look at the character Viconia, from Baldur's Gate.   She's romanceable, and *SPOILERS* if things go very very well, at the end Baldur's Gate 2's, you may actually get her to give up on being evil. *END SPOILERS*.
Though the writing is far from perfect, it might serve us well here. There's even a youtube with all the romance cuts for qui reference, but I recommend actually playing the game.

**Comment by CrocWorks** [comment]
Right, that's why I mentioned it may or may not be tied to affection. I realize that the more detailed this system is made, the better the mod will be, but it also increases the amount of work for whoever designs the system.

I guess one simpler way to do this would be just change the amount of morality points a conversation gives based on affection level. It probably could go the other way, too. As you said, Monika is a pretty confident person, so if certain dialogue options imply that Monika currently believes in one morality, and the player believes in another and picks it, then that would push Monika towards that morality but could also be at a small cost of affection.

I just see this as a way to flesh out the current affection system. Right now it seems pretty basic where you either "be nice to Monika" or "be mean to Monika" and it's too obvious how your dialogue choices are going to affect her opinion of you.

**Comment by ThePotatoGuy** [comment]
tl;dr, monika learns from the player's choices and changes her personality, right?

Would be an **extremely** complex system, but the payoff is pretty powerful. 

Very unlikely to happen soon.

**Comment by CrocWorks** [comment]
Yeah, pretty much. I don't know how complex the code for the affection system is, but if that could be copied and modified as a base for this one then that could help speed things along... although when I first wrote this, I was imagining Affection was just a single block of code that could be easily copy/pasted and then renamed. But I've started learning a little bit about coding and I'm guessing bits of affection code are scattered everywhere so that different things affect it?

**Comment by KEJONACHO** [comment]
Probably a big effort to come up with properly, but i still believe it's a good idea. But, to make things simpler, it could  be not a full system, like in affection, but instead just remembering certain answers from the player, like how they decide pro/against other dokies or small cruelties in general, or a number that increases/decreases based on dialogues the players repeat from her.
It can be great if we can make a fully fleshed-out system with it, though...

**Comment by ThePotatoGuy** [comment]
[closing until we look into this]

**Comment by CrocWorks** [comment]
Recently thought of a tangent and wanted to write it down somewhere. Perhaps Monika's morality starting point could be changed depending on which act you installed MAS from. If you installed during Act 4 without deleting her then she'd be closer to the cold side, before Act 3 would be more neutral, and after you delete her she'd be more remorseful.

**Comment by TheGlader2** [comment]
I do believe they are already working on this.
